File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/RoundEndedEvent.java
package dev.robocode.tankroyale.botapi.events;

import dev.robocode.tankroyale.botapi.BotResults;

/**
 * Event occurring when a round has just ended.
 */
@SuppressWarnings("unused")
public final class RoundEndedEvent implements IEvent {

    // The round number.
    private final int roundNumber;

    // The turn number.
    private final int turnNumber;

    // The accumulated bot results.
    private final BotResults results;

    /**
     * Initializes a new instance of the RoundEndedEvent class.
     *
     * @param roundNumber is the round number.
     * @param turnNumber  is the turn number.
     * @param results     is the accumulated bot results at the end of the round.
     */
    public RoundEndedEvent(int roundNumber, int turnNumber, BotResults results) {
        this.roundNumber = roundNumber;
        this.turnNumber = turnNumber;
        this.results = results;
    }

    /**
     * Returns the round number.
     *
     * @return The round number.
     */
    public int getRoundNumber() {
        return roundNumber;
    }

    /**
     * Returns the turn number.
     *
     * @return The turn number.
     */
    public int getTurnNumber() {
        return turnNumber;
    }

    /**
     * Returns the accumulated bot results at the end of the round.
     *
     * @return The accumulated bot results at the end of the round.
     */
    public BotResults getResults() {
        return results;
    }
}

File: bot_api_py/events/round_ended_event.py
from bot_api_py.bot_results import BotResults
from bot_api_py.events.abstract_event import IEvent


class RoundEndedEvent(IEvent):
    """Event occurring when a round has just ended."""

    def __init__(self, round_number: int, turn_number: int, results: BotResults):
        """Initializes a new instance of the RoundEndedEvent class.

        Args:
            round_number: The round number.
            turn_number: The turn number.
            results: The accumulated bot results at the end of the round.
        """
        self.round_number = round_number
        self.turn_number = turn_number
        self.results = results

    def get_round_number(self) -> int:
        """Returns the round number.

        Returns:
            The round number.
        """
        return self.round_number

    def get_turn_number(self) -> int:
        """Returns the turn number.

        Returns:
            The turn number.
        """
        return self.turn_number

    def get_results(self) -> BotResults:
        """Returns the accumulated bot results at the end of the round.

        Returns:
            The accumulated bot results at the end of the round.
        """
        return self.results